util/reggen/uvm_reg_base.sv.tpl

```diff
@@ -56,13 +86,24 @@ ${make_ral_pkg_window_class(dv_base_prefix, esc_if_name, window)}
     // registers
   % for r in rb.all_regs:
 <%
-      r0 = r.reg if isinstance(r, MultiRegister) else r
+      regs = []
+      if isinstance(r, MultiRegister):
+        if r.dv_compact:
+          regs.append(r.reg)
+        else:
+          regs.extend(r.regs)
+      else:
+        regs.append(r)
+%>\
+    % for r0 in regs:
+<%
       reg_type = gen_dv.rcname(esc_if_name, r0)
       inst_name = r0.name.lower()
-      count = r.count if isinstance(r, MultiRegister) else 1
+      count = len(r.regs) if isinstance(r, MultiRegister) and r.dv_compact else 1
```

Comment:
Since we've got the if/else stuff in the block above, maybe it would be clearer to move this there? That gets rid of the big ternary, which should make things a bit easier to understand.

Reply:
Indeed, moved it. Thanks